ios/MdocDataTransfer.swift

```diff
@@ -0,0 +1,195 @@
+import MdocDataModel18013
+import MdocDataTransfer18013
+import MdocSecurity18013
+import React
+import SwiftCBOR
+
+@objc(MdocDataTransfer)
+class MdocDataTransfer: RCTEventEmitter {
+    let ON_RESPONSE_SENT_EVENT: String = "onResponseSent"
+    let ON_REQUEST_RECEIVED_EVENT: String = "onRequestReceived"
+
+    var bleServerTransfer: MdocGattServer?
+    var resolver: RCTPromiseResolveBlock?
+    var rejector: RCTPromiseRejectBlock?
+
+    override func supportedEvents() -> [String]! {
+        return ["onResponseSent", "onRequestReceived"]
+    }
+
+    // NFC is not enabled on iOS
+    @objc
+    func enableNfc() {
+        return
+    }
+
+    @objc
+    func initialize() -> String {
+        guard bleServerTransfer == nil else {
+            return ""
+        }
+
+        do {
+            bleServerTransfer = try MdocGattServer(parameters: [
+                InitializeKeys.document_json_data.rawValue: [],
+                InitializeKeys.trusted_certificates.rawValue: [],
+                InitializeKeys.send_response_manually.rawValue: true,
+            ])
+            bleServerTransfer?.delegate = self
+        } catch {
+            return error.localizedDescription
+        }
+
+        return ""
+    }
+
+    @objc(startQrEngagement:_:)
+    func startQrEngagement(
+        resolve: @escaping RCTPromiseResolveBlock,
+        reject: @escaping RCTPromiseRejectBlock
+    ) {
+        resolver = resolve
+        rejector = reject
+
+        guard let bleServerTransfer = bleServerTransfer else {
+            self.reject(
+                MdocDataTransferError.BleGattServerNotInitialized
+                    .localizedDescription)
+            return
+        }
+
+        bleServerTransfer.performDeviceEngagement()
+    }
+
+    @objc(sendDeviceResponse:)
+    func sendDeviceResponse(deviceResponse: String) -> String {
+        guard let bleServerTransfer = bleServerTransfer,
+            var sessionEncryption = bleServerTransfer.sessionEncryption
+        else {
+            return MdocDataTransferError.BleGattServerNotInitialized
+                .localizedDescription
+        }
+
+        do {
+            let byteArray = deviceResponse.split(separator: ":").compactMap {
+                UInt8($0)
+            }
+            let cipherData = try sessionEncryption.encrypt(byteArray)
+            let sd = SessionData(cipher_data: cipherData, status: 20)
+            try bleServerTransfer.sendResponse(
+                Data(sd.encode(options: CBOROptions())))
```

Comment:
Is this BLE session encryption or mdoc specific encryption? Does the android library also handle this? or is that handled outside of this library (e.g. in mdoc library?)

Reply:
This is handled exactly like the android side. I can also move this to the library.